src/pluginplay/printing/mermaid.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 NWChemEx-Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "module/module_class.hpp"
+#include "module_manager/module_manager_class.hpp"
+#include <pluginplay/pluginplay.hpp>
+
+void print_submods(const std::string module,
+                   const pluginplay::ModuleManager& mm, int level = 0) {
+    auto mm_module = mm.at(module);
+    auto submods   = mm_module.submods();
+
+    for(const auto& [key, value] : submods) {
+        std::cout << "Submod ID: " << key << std::endl;
+        if(value.has_module() == false) {
```

Comment:
Just to be clear, I wouldn't skip edges of the graph that only have a source. Users will want to see this in the documentation (it tells them where they still need to select submodules). I don't know if mermaid supports sources without sinks (just in case you haven't seen the terminology when a graph node is an arrow, "sources" are  graph node the edge comes from and "sinks" are the nodes the arrow points to), or if we'll have to have it point to "Void" (or something).